### PR TITLE
CNFT2-1481 classic preview

### PR DESCRIPTION
**Pull request by mpeels**
## Description

In the modernized page builder, a link needs to exist to allow the user to view a page in the Classic display. This PR adds a `Classic preview` button as well as handles the routing to load the Classic preview page.

If the user clicks the `Edit Page` button on the Classic page, the user will be redirected back to the Modernized edit page. Other buttons are not yet routed from Classic.

## Tickets

* [CNFT2-1481](https://cdc-nbs.atlassian.net/browse/CNFT2-1481)

## Steps to verify

1. Restart nbs-gateway, question-bank applications
2. Navigate to [NBS](http://localhost:8080/nbs/login)
3. Log in as `superuser`
4. Click System Management -> Page Management -> Manage Pages
5. Click and Investigation page from the table
6. Click the new `Eye`con 
7. Verify a new tab is opened and the correct page is loaded in the Classic preview
8. Click the `Edit Page` button
9. Verify the Modernized edit page is loaded

Nov-22-2023 17-51-50

[CNFT2-1481]: https://cdc-nbs.atlassian.net/browse/CNFT2-1481?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/modernization-ui/src/apps/page-builder/pages/EditPage/EditPageHeader/EditPageHeader.tsx [hunk 1]
```diff
@@ -39,6 +39,13 @@ export const EditPageHeader = ({ page, handleSaveDraft }: PageProps) => {
                         Save draft
                     </Button>
                 )}
+                <LinkButton
+                    href={`/nbs/page-builder/api/v1/pages/${page.id}/preview`}
+                    label="open a preview of the current page"
+                    rel="noreferrer"
```
Comment (adamloup-enquizit) [inline apps/modernization-ui/src/apps/page-builder/pages/EditPage/EditPageHeader/EditPageHeader.tsx:45]: question:  I wonder if these should be the defaults?

File: apps/nbs-gateway/src/test/java/gov/cdc/nbs/gateway/pagebuilder/EditPageRouteConfigurationDisabledTest.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  WebTestClient webClient;
+
+  @Autowired
+  RouteLocator pagebuilderManagePagesConfig;
+
+  @Autowired
+  PageBuilderService service;
+
+  @Test
+  void should_contain_route() {
```
Comment (adamloup-enquizit) [inline apps/nbs-gateway/src/test/java/gov/cdc/nbs/gateway/pagebuilder/EditPageRouteConfigurationDisabledTest.java:46]: nit: I think test that it doesn't route is covering this case.  The name of the routes only help with logging.

File: apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/page/classic/ClassicCreateTemplatePreparer.java [hunk 3]
```diff
@@ -1,11 +1,11 @@
-package gov.cdc.nbs.questionbank.page.template;
+package gov.cdc.nbs.questionbank.page.classic;
```
Comment (adamloup-enquizit) [inline apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/page/classic/ClassicCreateTemplatePreparer.java:1]: nit:  I was torn too, it's `classic` but it's `template` where does it go?

File: apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/page/classic/preview/ClassicPagePreviewRedirector.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  ResponseEntity<Void> view(@PathVariable("page") final long page) {
+    ReturningPageCookie pageCookie = new ReturningPageCookie(String.valueOf(page));
+    preparer.prepare();
+
+    URI location = UriComponentsBuilder.fromPath(PREVIEW)
+        .queryParam("from", "L")
+        .queryParam("method", "viewPageLoad")
+        .queryParam("waTemplateUid", page)
+        .build()
+        .toUri();
```
Comment (adamloup-enquizit) [inline apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/page/classic/preview/ClassicPagePreviewRedirector.java:39]: nit: There is a `toUriString()` method also.

File: apps/question-bank/src/test/java/gov/cdc/nbs/questionbank/page/classic/preview/ClassicPagePreviewRedirectorTest.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+class ClassicPagePreviewRedirectorTest {
+
+  @Mock
+  private ClassicPreviewPagePreparer preparer;
+
+  @InjectMocks
+  private ClassicPagePreviewRedirector redirector;
+
+  @Test
+  void should_redirect_to_preview() {
```
Comment (adamloup-enquizit) [inline apps/question-bank/src/test/java/gov/cdc/nbs/questionbank/page/classic/preview/ClassicPagePreviewRedirectorTest.java:23]: nit:  should the cookie be tested as part of the feature test?

**Comment by adamloup-enquizit** [comment]
The `Return-Page` cookie is not being removed when navigating back to modernization.  could be tricky to do with the generic redirector.  Maybe there can be a configuration that specifies which cookies to remove on the return trip?

File: apps/modernization-ui/src/apps/page-builder/pages/EditPage/EditPageHeader/EditPageHeader.tsx (hunk 1 above)
Comment (mpeels) [reply apps/modernization-ui/src/apps/page-builder/pages/EditPage/EditPageHeader/EditPageHeader.tsx:45]: I think so.

File: apps/nbs-gateway/src/test/java/gov/cdc/nbs/gateway/pagebuilder/EditPageRouteConfigurationDisabledTest.java (hunk 2 above)
Comment (mpeels) [reply apps/nbs-gateway/src/test/java/gov/cdc/nbs/gateway/pagebuilder/EditPageRouteConfigurationDisabledTest.java]: Resolution: remove unnecessary test

**Review by adamloup-enquizit — APPROVED** [review]
📃 🍪

**Pull request merged**